src/flags.zig

```diff
@@ -206,21 +281,108 @@ fn parse_flag_value_int(comptime T: type, comptime flag: []const u8, value: [:0]
     };
 }
 
-fn field_to_flag(comptime field: []const u8) []const u8 {
+pub const ByteSize = struct { bytes: u64 };
+fn parse_value_size(comptime flag: []const u8, value: []const u8) ByteSize {
+    comptime assert((flag[0] == '-' and flag[1] == '-') or flag[0] == '<');
+
+    const units = .{
+        .{ &[_][]const u8{ "TiB", "tib", "TB", "tb" }, 1024 * 1024 * 1024 * 1024 },
+        .{ &[_][]const u8{ "GiB", "gib", "GB", "gb" }, 1024 * 1024 * 1024 },
+        .{ &[_][]const u8{ "MiB", "mib", "MB", "mb" }, 1024 * 1024 },
+        .{ &[_][]const u8{ "KiB", "kib", "KB", "kb" }, 1024 },
+    };
+
+    const unit: struct { suffix: []const u8, scale: u64 } = unit: inline for (units) |unit| {
+        const suffixes = unit[0];
+        const scale = unit[1];
+        for (suffixes) |suffix| {
+            if (std.mem.endsWith(u8, value, suffix)) {
+                break :unit .{ .suffix = suffix, .scale = scale };
+            }
+        }
+    } else break :unit .{ .suffix = "", .scale = 1 };
+
+    assert(std.mem.endsWith(u8, value, unit.suffix));
+    const value_numeric = value[0 .. value.len - unit.suffix.len];
+
+    const amount = std.fmt.parseUnsigned(u64, value_numeric, 10) catch |err| {
+        fatal("{s}: expected a size, but found '{s}' ({s})", .{
+            flag, value, switch (err) {
+                error.Overflow => "value too large",
+                error.InvalidCharacter => "invalid digit",
+            },
+        });
+    };
+
+    var bytes: u64 = undefined;
+    if (@mulWithOverflow(u64, amount, unit.scale, &bytes)) {
+        fatal("{s}: expected a size, but found '{s}' (value too large)", .{
+            flag, value,
+        });
+    }
+    return ByteSize{ .bytes = bytes };
+}
+
+test parse_value_size {
+    const kib = 1024;
+    const mib = kib * 1024;
+    const gib = mib * 1024;
+    const tib = gib * 1024;
+
+    const cases = .{
+        .{ 0, "0" },
+        .{ 1, "1" },
+        .{ 140737488355328, "140737488355328" },
+        .{ 140737488355328, "128TiB" },
+        .{ 1 * tib, "1TiB" },
+        .{ 10 * tib, "10tib" },
+        .{ 100 * tib, "100TB" },
+        .{ 1000 * tib, "1000tb" },
+        .{ 1 * gib, "1GiB" },
+        .{ 10 * gib, "10gib" },
+        .{ 100 * gib, "100GB" },
+        .{ 1000 * gib, "1000gb" },
+        .{ 1 * mib, "1MiB" },
+        .{ 10 * mib, "10mib" },
+        .{ 100 * mib, "100MB" },
+        .{ 1000 * mib, "1000mb" },
+        .{ 1 * kib, "1KiB" },
+        .{ 10 * kib, "10kib" },
+        .{ 100 * kib, "100KB" },
+        .{ 1000 * kib, "1000kb" },
```

Comment:
NB: I intentionally removed support for spaces in the arguments. `--size='6 Gib'` feels very error-prone, if it is printed somewhere and then copy-pasted.